internal/pkg/util/util.go

```diff
@@ -382,3 +382,49 @@ func IncrementIPv4(start string, inc uint) string {
 	ipv4_new := net.IPv4(v4_o0, v4_o1, v4_o2, v4_o3)
 	return ipv4_new.String()
 }
+
+func HostnameToV4(hostname string) (ipv4 net.IP, err error) {
+	if len(hostname) == 0 {
+		return ipv4, fmt.Errorf("hostname cannot be empty")
+	}
+	ips, err := net.LookupIP(hostname)
+	if err != nil {
+		return ipv4, err
+	}
+	for _, ip := range ips {
+		if v4 := ip.To4(); v4 != nil {
+			return v4, nil
+		}
+	}
+	return ipv4, fmt.Errorf("could not get ipv4 from hostname")
+}
+
+func GetFirstIPv4(iface *net.Interface) (ipv4 net.IP, err error) {
```

Comment:
Just to be sure I'm following, this function is returning the IPv4 address of a specified network interface configuration?

Is it "the first" just incase the device has multiple IP addresses configured for that interface (e.g. not aliases)?

If it has multiple addresses, do we want to figure that out, and then require someone to define the ipaddr instead?

Reply:
Yes, the first IPv4 is returned in the event the device has multiple addresses configured...

I'm unsure of how `net.Interface.Addrs()` treats alias addresses, so it's definitely something to consider.

Here's the way `util.GetFirstIPv4()` works functionally:

1. Get all the addresses that exist on the inteface. (`addrs := []net.Addr`)
2. For each value in `addrs`, if the criteria below is met, the IPv4 in question is returned.

- `ip := value.(type)` must be either `*net.IPAddr` or `*net.IPNet`
- Is `r := ip.IP.To4()` `nil`? If yes, do nothing and check the next address in the slice. If no, return `r`.


...I think that prompting the user for input could be helpful, but is that what the community wants? What is the best way to prompt a user for input when `wwctl server start` sends the process to the background?

Reply:
I like that flow, with one potential addition...

If someone specifies the `ipaddr` in the configuration file, what if it matches the second or third address in the `[]net.Addr` array? That should work but I believe it would fail in the current implementation. 

Thoughts?

Reply:
It doesn't seem to fail the way you described. If `ipaddr` is specified, `device_name` is ignored. I just tested it with the config file shown [here](https://ghostbin.com/gGZCg).

`eth0` looks like [this](https://ghostbin.com/T7KIe).